yt_dlp/networking/common.py

```diff
@@ -216,6 +216,22 @@ def read(self, amt: int = None):
         raise NotImplementedError
 
 
+class SimpleHTTPResponse(HTTPResponse):
```

Comment:
I like this. might port it back up since we can use it for the base of the backend specific http responses. 

Alternatively it might just be better to make HTTPResponse take data itself rather than leaving to subclasses :thinking: